core/vdbe/execute.rs

```diff
@@ -3759,14 +3764,41 @@ pub fn op_delete(
     Ok(InsnFunctionStepResult::Step)
 }
 
+pub fn op_idx_delete(
+    program: &Program,
+    state: &mut ProgramState,
+    insn: &Insn,
+    pager: &Rc<Pager>,
+    mv_store: Option<&Rc<MvStore>>,
+) -> Result<InsnFunctionStepResult> {
+    let Insn::IdxDelete {
+        cursor_id,
+        start_reg,
+        num_regs,
+    } = insn
+    else {
+        unreachable!("unexpected Insn {:?}", insn)
+    };
+    let record = make_record(&state.registers, start_reg, num_regs);
+    {
+        let mut cursor = state.get_cursor(*cursor_id);
+        let cursor = cursor.as_btree_mut();
+        return_if_io!(cursor.seek(SeekKey::IndexKey(&record), SeekOp::EQ));
+        return_if_io!(cursor.delete());
```

Comment:
`IdxDelete` should fail with CorruptError if the key to be deleted is not found:

> If P5 is not zero, then raise an SQLITE_CORRUPT_INDEX error if no matching index entry is found. This happens when running an UPDATE or DELETE statement and the index entry to be updated or deleted is not found. For some uses of [IdxDelete](https://www.sqlite.org/opcode.html#IdxDelete) (example: the EXCEPT operator) it does not matter that no matching entry is found. For those cases, P5 is zero. Also, do not raise this (self-correcting and non-critical) error if in writable_schema mode.

https://www.sqlite.org/opcode.html

Reply:
Ah thanks!